source/base/subscriptor.cc

```diff
@@ -126,18 +126,17 @@ Subscriptor::operator=(Subscriptor &&s) noexcept
 
 
 
-template <>
 void
-Subscriptor::subscribe<const char *>(std::atomic<bool> *const validity,
-                                     const char *             id) const
+Subscriptor::subscribe(std::atomic<bool> *const validity,
+                       const std::string &      id) const
 {
   std::lock_guard<std::mutex> lock(mutex);
 
   if (object_info == nullptr)
     object_info = &typeid(*this);
   ++counter;
 
-  const char *const name = (id != nullptr) ? id : unknown_subscriber;
+  const std::string name = (!id.empty()) ? id : unknown_subscriber;
```

Comment:
Its not critical, but would you mind switching the order on these so that we don't need the negation? i.e.,
```cpp
id.empty() ? unknown_subscriber : id;
```
Furthermore, could we use a value here and move it into the counter map below?